client/src/app/containers/Game/saga.js

```diff
@@ -0,0 +1,174 @@
+import { all, put, select, takeLatest, call } from 'redux-saga/effects';
+import axios from 'axios';
+import { selectCurrentUser } from '../../../auth/selectors';
+import { selectCurrentGameID, selectCurrentAction } from './selectors';
+import { actions as gameActions } from './slice';
+import { actions as alertActions } from '../Alert/slice';
+import { alertTypes } from '../Alert/types';
+
+export function* handleExitGame({ payload: { history } }) {
+  yield call(history.push, '/home');
+}
+
+export function* handleGoToGame({ payload: { id, history } }) {
+  if (!id) {
+    yield put(
+      alertActions.addAlert('No id in handleGoToGame', alertTypes.error),
+    );
+    return;
+  }
+
+  const currentUser = yield select(selectCurrentUser);
+
+  // select the id being used to login from the state
+  try {
+    const res = yield axios.get(`/game/${id}?player=${currentUser.id}`);
+    yield put(gameActions.gameRetrieved({ data: res.data }));
+    yield call(history.push, '/game');
+  } catch (err) {
+    yield put(
+      alertActions.addAlert(
+        `something bad happened... ${err}`,
+        alertTypes.error,
+      ),
+    );
+  }
+}
+
+export function* handleRefreshCurrentGame({ payload: { id, history } }) {
+  const currentUser = yield select(selectCurrentUser);
+
+  try {
+    const res = yield axios.get(`/game/${id}?player=${currentUser.id}`);
+    yield put(gameActions.gameRetrieved({ data: res.data }));
+  } catch (err) {
+    yield put(alertActions.addAlert(err.response.data, alertTypes.error));
+  }
+}
+
+const cardToGolangCard = c => {
+  const magicMap = {
+    Spades: 0,
+    Clubs: 1,
+    Diamonds: 2,
+    Hearts: 3,
+  };
+  return {
+    s: magicMap[c.suit],
```

Comment:
Ope, don't like this... Why do we need to convert anything? If we're converting anything for the backend, the backend should handle it

Reply:
🤷  I don't mind magic maps on the client; but I'm no hard-core client dev